File: ros2-py-outputs/agr/autonomous_ground_robot/imu_driver_node.py
"""ImuDriver — node implementation. EDIT ME.

Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
(generation-gap pattern): this file is where demo/application logic lives.
The architectural wiring (endpoints, parameters, lifecycle transitions) is in
imu_driver_node_base.py, which is regenerated on every pipeline run.
"""
import rclpy

from .imu_driver_node_base import ImuDriverBase


class ImuDriver(ImuDriverBase):
    """imu application logic."""

    # Override on_configure_hook / on_activate_hook / on_deactivate_hook here.


def main(args=None):
    rclpy.init(args=args)
    node = ImuDriver()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: ros2-cpp-outputs/agr/src/imu_driver_node.cpp
// ImuDriver — node implementation. EDIT ME.
//
// Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
// (generation-gap pattern): this file is where demo/application logic lives.
// The architectural wiring is in imu_driver_node_base.hpp/.cpp, regenerated
// on every pipeline run.
#include "autonomous_ground_robot/imu_driver_node_base.hpp"

class ImuDriver : public ImuDriverBase
{
public:
  using ImuDriverBase::ImuDriverBase;

protected:
  // Override on_configure_hook / on_activate_hook / on_deactivate_hook here.
};

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<ImuDriver>();
  rclcpp::spin(node->get_node_base_interface());
  rclcpp::shutdown();
  return 0;
}
